Building DEBUG=1 fails with gcc10 changes and -flto enabled

Doing a debug build after the gcc10 PR now fails. The failure can be avoided by removing `-flto`, but that can cause firmware overflow. It has to do with inlining. I think I tried to remove `-fno-inline-functions` just to help diagnose this, but I think that didn't change things (not sure -- I tried many things).

```
../../py/vm.c: In function 'mp_execute_bytecode':
../../supervisor/shared/translate.c:125:66: error: inlining failed in call to 'always_inline' 'translate': optimization level attribute mismatch
  125 | inline __attribute__((always_inline)) const compressed_string_t* translate(const char* original) {
      |                                                                  ^
../../py/vm.c:257:44: note: called from here
  257 |                             mp_obj_t obj = mp_obj_new_exception_msg(&mp_type_NameError, translate("local variable referenced before assignment"));
      |                                            ^
../../supervisor/shared/translate.c:125:66: error: inlining failed in call to 'always_inline' 'translate': optimization level attribute mismatch
  125 | inline __attribute__((always_inline)) const compressed_string_t* translate(const char* original) {
      |                                                                  ^
../../py/vm.c:1144:35: note: called from here
 1144 |                             obj = mp_obj_new_exception_msg(&mp_type_RuntimeError, translate("no active exception to reraise"));
      |                                   ^
../../supervisor/shared/translate.c:125:66: error: inlining failed in call to 'always_inline' 'translate': optimization level attribute mismatch
  125 | inline __attribute__((always_inline)) const compressed_string_t* translate(const char* original) {
      |                                                                  ^
../../py/vm.c:1286:36: note: called from here
 1286 |                     mp_obj_t obj = mp_obj_new_exception_msg(&mp_type_NotImplementedError, translate("byte code not implemented"));
      |                                    ^
lto-wrapper: fatal error: arm-none-eabi-gcc returned 1 exit status
```

--- a/py/circuitpy_defns.mk
+++ b/py/circuitpy_defns.mk
@@ -58,6 +58,14 @@ BASE_CFLAGS = \
 #        -H
 
 
+# Set a global CIRCUITPY_DEBUG flag.
+# Don't just call it "DEBUG": too many libraries use plain DEBUG.
+ifneq ($(DEBUG),)
+CFLAGS += -DCIRCUITPY_DEBUG=$(DEBUG)
+else
+CFLAGS += -DCIRCUITPY_DEBUG=0
+endif
+
 ###
 # Handle frozen modules.
 

--- a/supervisor/shared/translate.c
+++ b/supervisor/shared/translate.c
@@ -122,7 +122,12 @@ char* decompress(const compressed_string_t* compressed, char* decompressed) {
     return decompressed;
 }
 
-inline __attribute__((always_inline)) const compressed_string_t* translate(const char* original) {
+inline
+// gcc10 -flto has issues with this being always_inline for debug builds.
+#if CIRCUITPY_DEBUG < 1
+ __attribute__((always_inline))
+#endif
+const compressed_string_t* translate(const char* original) {
     #ifndef NO_QSTR
     #define QDEF(id, str)
     #define TRANSLATION(id, firstbyte, ...) if (strcmp(original, id) == 0) { static const compressed_string_t v = { .data = firstbyte, .tail = { __VA_ARGS__ } }; return &v; } else

